### PR TITLE
fix(install): setup logic, paq instructions

### PR DESCRIPTION
**Pull request by matu3ba**
telescope may not be configured in the package file: move extension loading to separate line

### CODE REVIEW

**Comment by matu3ba** [comment]
You dont use plenary.nvim functionality of telescope, correct?

**Comment by delphinus** [comment]
Yes, telescope-z needs both of them to load telescope itself. I've illustrated the minimum settings, so it lacks to use it in real as you noticed.

File: README.md [hunk 1]
```diff
@@ -8,15 +8,15 @@
...
-    'nvim-telescope/telescope-z.nvim',
-  },
-  config = function()
-    require'telescope'.load_extension'z'
-  end,
-}
+-- packer.nvim:
+use{ 'nvim-telescope/telescope-z.nvim', requires = { 'nvim-telescope/telescope.nvim' } }
+...
+require'telescope'.load_extension'z'
```
Comment (delphinus) [inline README.md:14]:
```suggestion
use{
  'nvim-telescope/telescope.nvim',
  requires = {
    {'nvim-lua/plenary.nvim'},
    {'nvim-lua/popup.nvim'},
    {'nvim-telescope/telescope-z.nvim'},
  },
  config = function()
    require'telescope'.load_extension'z'

    // ... other telescope settings ...
  end,
}
```

With packer.nvim in usual, we use `config` section to write plugin-specific settings.

Comment (delphinus) [reply README.md:14]: It is also good to add requirements for your paq.nvim example.

**Comment by matu3ba** [comment]
@delphinus is it okay like this?

**Comment by delphinus** [comment]
@matu3ba Sorry for late.

I think plugin-specific settings should be included in `config` section when using packer.nvim (as I mentioned above). How do you think?

**Comment by matu3ba** [comment]
> @matu3ba Sorry for late.

Sorry to be also late.

> I think plugin-specific settings should be included in `config` section when using packer.nvim (as I mentioned above). How do you think?

Would this break stuff, when you do not setup telescope beforehand?

**Comment by delphinus** [comment]
`config` is the section just for your concern. In default of packer.nvim, plugins are loaded at the first, and then Neovim executes `config` sections.

https://github.com/wbthomason/packer.nvim#specifying-plugins

> config = string or function, -- Specifies code to run after this plugin is loaded.

So calling `load_extentions()` does not break anything. Use `config`.

----

FYI: When you want to execute codes _before_ loading plugins, you can use `setup` instead of `config`.

**Comment by matu3ba** [comment]
@delphinus Github has a hardcap at 80 characters for code. Is this okay like this or do you prefer a separate line for the keybinding code?

**Comment by delphinus** [comment]
No problem. It is easy enough to read. Thank you for contribution!

**Pull request merged**